src/utils/services/authHandler.js

```diff
@@ -0,0 +1,33 @@
+import axios from "axios";
+import { actionType } from "../../constants";
+
+const authHandler = async (credentials, authAction, authDispatch) => {
+  try {
+    const response = await axios.post(`/api/auth/${authAction.toLowerCase()}`, {
+      ...credentials,
+    });
+    const data = {
+      userInfo:
+        authAction === "LOGIN"
+          ? { ...response.data.foundUser }
+          : { ...response.data.createdUser },
+      token: response.data.encodedToken,
+    };
+
+    localStorage.setItem("data", JSON.stringify(data));
+    if (response.status === 201 || response.status === 200)
+      authDispatch({
```

Comment:
* You can extract the localStorage as a separate function
* Also you can do something like this `response.status === 201 || 200`

Reply:
yes, thank you jowel, i refactored the code now

Reply:
> yes, thank you jowel, i refactored the code now

Correction `response.status === 201 || 200` this code will not work, instead your previous code will work fine i.e `response.status === 201 || response.status === 200`. Sorry for wrong information.